include/glow/Runtime/HostManager/HostManager.h

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_RUNTIME_HOSTMANAGER_H
+#define GLOW_RUNTIME_HOSTMANAGER_H
+#include "glow/Graph/Graph.h"
+#include "glow/Runtime/Executor.h"
+#include "glow/Runtime/Provisioner.h"
+#include "glow/Runtime/RuntimeTypes.h"
+
+#include <llvm/ADT/StringRef.h>
+
+#include <atomic>
+#include <map>
+#include <unordered_map>
+#include <vector>
+
+namespace glow {
+namespace runtime {
+
+/// Callback signalling the result of running a function.
+using ResultCBTy = std::function<void(
+    runtime::RunIdentifierTy, runtime::ResultCode, std::unique_ptr<Context>)>;
+
+class DeviceManager;
+class Context;
+class DeviceConfig;
+class HostManager final : std::enable_shared_from_this<HostManager> {
+  /// Count of current in-flight networks being run.
+  std::atomic<size_t> activeRequestCount_{0};
+
+  /// Count of total requests, this is used as a run ID.
```

Comment:
Why is this atomic? Are you expecting multithreaded access to methods that mutate this field? I don't see any comments about this in the method that accesses the field.

Reply:
Will add comments to indicate multi-thread expectations.